src/software/networking/threaded_unix_listener.py

```diff
@@ -0,0 +1,114 @@
+import socketserver
+import time
+import base64
+import os
+from google.protobuf import text_format
+from google.protobuf.any_pb2 import Any
+from threading import Thread
+import queue
+
+
+class ThreadedUnixListener:
+    def __init__(
+        self, unix_path, proto_class, max_buffer_size=3, convert_from_any=True
+    ):
+
+        """Receive protobuf over unix sockets and buffers them
+
+        :param unix_path: The unix path to receive the new protobuf to plot
+        :param proto_class: The type of protobuf we expect to receive
+        :param max_buffer_size: The size of the buffer
+        :param convert_from_any: Convert from any
+
+        """
+
+        # cleanup the old path if it exists
+        try:
+            os.remove(unix_path)
+        except:
+            pass
+
+        self.server = socketserver.UnixDatagramServer(
+            unix_path,
+            handler_factory(proto_class, self.__buffer_protobuf, convert_from_any),
+        )
+        self.stop = False
+
+        self.unix_path = unix_path
+        self.proto_buffer = queue.Queue(max_buffer_size)
+
+        self.thread = Thread(target=self.start)
+        self.thread.start()
+
+    @property
+    def buffer(self):
+        return self.proto_buffer
+
+    def maybe_pop(self):
+        """Pop from the buffer if a new packet exists. If not just return None
+
+        :return: proto if exists, else None
+        :rtype: proto or None
+
+        """
+        try:
+            return self.proto_buffer.get_nowait()
+        except queue.Empty as empty:
+            return None
+
+    def __buffer_protobuf(self, proto):
+        """Buffer the protobuf, and raise a warning if we overrun the buffer
+
+        :param proto: The protobuf to buffer
+        :raises: Warning
+
+        """
+        try:
+            self.proto_buffer.put_nowait(proto)
+        except queue.Full as queue_full:
+            print("buffer overrun for {}".format(self.unix_path))
```

Comment:
I'm getting a bunch of these running locally, is that expected? It seems kind of spammy

Reply:
mainly World and Obstacle and I guess it doesn't happen all the time so not blocking but something to keep in mind